### PR TITLE
Fix: AlephClient class could not use unix sockets

### PR DESCRIPTION
**Pull request by hoh**

### CODE REVIEW

File: src/aleph/sdk/client.py [hunk 1]
```diff
@@ -443,9 +444,35 @@ class AlephClient:
...
+
+        connector: Optional[aiohttp.UnixConnector]
+        if api_unix_socket:
+            connector = aiohttp.UnixConnector(path=api_unix_socket)
+        elif api_unix_socket is None and settings.API_UNIX_SOCKET:
+            connector = aiohttp.UnixConnector(path=settings.API_UNIX_SOCKET)
+        else:
+            connector = None
+
+        if timeout:
```
Comment (odesenfans) [inline src/aleph/sdk/client.py:468]: I prefer a good old `kwargs = {"timeout": timeout}` in this case to reduce duplication.

File: src/aleph/sdk/client.py [hunk 2]
```diff
@@ -443,9 +444,35 @@ class AlephClient:
...
     http_session: aiohttp.ClientSession
 
-    def __init__(self, api_server: str):
-        self.api_server = api_server
-        self.http_session = aiohttp.ClientSession(base_url=api_server)
+    def __init__(
+        self,
+        api_server: Optional[str],
+        timeout: Optional[aiohttp.ClientTimeout] = None,
+        api_unix_socket: Union[None, str, Literal[False]] = None,
```
Comment (odesenfans) [inline src/aleph/sdk/client.py:451]: This typing is a bit weird. In such cases I prefer to add a second option, like `allow_unix_socket: bool`.

File: src/aleph/sdk/client.py [hunk 3]
```diff
@@ -443,9 +444,35 @@ class AlephClient:
...
+        elif settings.API_HOST:
+            self.api_server = settings.API_HOST
+        else:
+            raise ValueError("Missing API host")
+
+        connector: Optional[aiohttp.UnixConnector]
+        if api_unix_socket:
+            connector = aiohttp.UnixConnector(path=api_unix_socket)
+        elif api_unix_socket is None and settings.API_UNIX_SOCKET:
+            connector = aiohttp.UnixConnector(path=settings.API_UNIX_SOCKET)
```
Comment (odesenfans) [inline src/aleph/sdk/client.py:464]: Nitpick: little bit of duplication here, maybe extract the socket path before spawning the `aiohttp.UnixConnector` instance.

File: src/aleph/sdk/client.py [hunk 4]
```diff
@@ -443,9 +444,35 @@ class AlephClient:
...
-    def __init__(self, api_server: str):
-        self.api_server = api_server
-        self.http_session = aiohttp.ClientSession(base_url=api_server)
+    def __init__(
+        self,
+        api_server: Optional[str],
+        timeout: Optional[aiohttp.ClientTimeout] = None,
+        api_unix_socket: Union[None, str, Literal[False]] = None,
+    ):
+        if api_server:
```
Comment (odesenfans) [inline src/aleph/sdk/client.py:453]: Could you add checks to verify that the options specified by the user make sense? Ex: they didn't specify `api_server` and `api_unix_socket` at the same time.

**Review by odesenfans — CHANGES_REQUESTED** [review]

File: src/aleph/sdk/client.py (hunk 4 above)
Comment (hoh) [reply src/aleph/sdk/client.py:453]: api_server is required by the HTTP protocol, even when using Unix sockets.

File: src/aleph/sdk/client.py (hunk 1 above)
Comment (hoh) [reply src/aleph/sdk/client.py]: I propose another approach using `functools.partial`.

**Pull request merged**